src/core/iframeMessageProtection.ts

```
@@ -0,0 +1,99 @@
+import { assert } from "tsafe/assert";
+import { asymmetricEncrypt, asymmetricDecrypt, generateKeys } from "../tools/asymmetricEncryption";
+import { type AuthResponse } from "./AuthResponse";
+
+const setItem_real = Storage.prototype.setItem;
+
+const SESSION_STORAGE_PREFIX = "oidc-spa_iframe_authResponse_publicKey_";
+
+export function preventSessionStorageSetItemOfPublicKeyByThirdParty() {
+    const setItem_protected = function setItem(this: any, key: string, value: string): void {
+        if (this !== sessionStorage) {
+            return setItem_real.call(this, key, value);
```

Biome (2.1.2)

[error] The function should not return a value because its return type is void.

The function is here:

'void' signals the absence of value. The returned value is likely to be ignored by the caller.

(lint/correctness/noVoidTypeReturn)